test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -39,6 +39,22 @@ class test_ucp_sockaddr : public ucp_test {
         UCS_TEST_SKIP_R("No interface for testing");
     }
 
+    void inaddr_any_addr(struct sockaddr_in *addr, in_port_t port)
+    {
+        memset(addr, 0, sizeof(struct sockaddr_in));
+        addr->sin_family      = AF_INET;
+        addr->sin_addr.s_addr = INADDR_ANY;
+        addr->sin_port        = port;
+    }
+
+    void start_listener(const struct sockaddr* addr)
+    {
+        ucs_status_t status = receiver().listen(addr, sizeof(*addr));
+        if (status == UCS_ERR_INVALID_ADDR) {
+            UCS_TEST_SKIP_R("cannot listen to " + ucs::sockaddr_to_str(addr));
```

Comment:
then error, probably, should be `UCS_ERR_UNSUPPORTED`?

Reply:
then will need to add a third return status from ucp_listener_create() but handle it the same way as UCS_ERR_INVALID_PARAM. you think it's required?

Reply:
I still don't understand why UCS_ERR_INVALID_PARAM if address is valid but no transport which support it.

Reply:
I found that `ucp_wireup_select_sockaddr_transport` returns `UCS_ERR_UNREACHABLE`, maybe this one also the best here.
How are you getting `UCS_ERR_INVALID_PARAM` in this test? `addr == NULL` or `UCP_LISTENER_PARAM_FIELD_SOCK_ADDR` is not set?

Reply:
yes. i think it's also possible to say that if we cannot listen, then the given tl value is invalid. but i dont think it's critical, it just adds a third return status to check. 
@yosefe what do you think? should we update https://github.com/openucx/ucx/blob/master/src/ucp/core/ucp_listener.c#L170 to UCS_ERR_UNSUPPORTED or UCS_ERR_UNREACHABLE?

Reply:
Let's imagine that user may don't specialize TLS at all and he has some HW which has addressable interface but UCX does not support it for listening. Is this address invalid parameter? How user should check it programmatically?

Reply:
if param is OK, but address not supported, should be unreachable
INVALID param is when the user has a way to avoid the situation, in this it's not possible

Reply:
ok, will update to UCS_ERR_UNSUPPORTED then.

Reply:
should be ERR_UNREACHABLE